pkg/envoy/sds/builder.go

```diff
@@ -53,62 +57,59 @@ func (b *sdsBuilder) SetTrustDomain(trustDomain string) *sdsBuilder {
 }
 
 func (b *sdsBuilder) SetServiceIdentitiesForService(serviceIdentitiesForServices map[service.MeshService][]identity.ServiceIdentity) *sdsBuilder {
-	b.serviceIdentitiesForOutboundServices = serviceIdentitiesForServices
+	b.identitiesForSecrets = make(map[string][]identity.ServiceIdentity)
+	for svc, serviceIdentities := range serviceIdentitiesForServices {
+		b.identitiesForSecrets[secrets.NameForUpstreamService(svc)] = serviceIdentities
+	}
 	return b
 }
 
+// TODO(2683): Remove this function once we use the snapshot cache, and remove requestedCerts.
+func (b *sdsBuilder) shouldBuild(secretName string) bool {
```

Comment:
Could you add a comment stating what this function does.

Reply:
will do!